tests/jsmarionette/client/marionette-client/lib/marionette/drivers/promises.js

```diff
@@ -0,0 +1,97 @@
+var Tcp = require('./tcp');
+var Response = require('../message').Response;
+var Command = require('../message').Command;
+
+Promises.Socket = Tcp.Socket;
+
+var DEFAULT_HOST = 'localhost';
+var DEFAULT_PORT = 2828;
+
+function Promises(options) {
+  if (!options) {
+    options = {};
+  }
+
+  this._lastId = 0;
+  this.host = options.host || DEFAULT_HOST;
+  this.port = options.port || DEFAULT_PORT;
+
+  this.connectionTimeout = options.connectionTimeout || 2000;
```

Comment:
Sorry, these leftovers are from the hack around I was doing trying to understand the protocol and the packet/message flow. The b2g-desktop I was running was from around two months ago, and after I rebuild it I see the protocol changed to 3.